### ISSUE TITLE
Error when using attribute_driven recipe

### ISSUE THREAD

**Issue by sterago** [hectcastro/chef-collectd#49]
Hi, I am getting this error when I use the attribute_driven recipe:
```
No resource or method named `collectd_plugin' for `Chef::Recipe "attribute_driven"'
```
I have noticed that `collectd_plugin` is not defined anywhere in this cookbook. Is it perhaps something defined in some other cookbook? Should a dependency be added if that's the case?

I am using version 1.2.0

thanks

**Comment by hectcastro** [comment]
Are you also including the `collectd` recipe along with `attribute_driven`?

**Comment by sterago** [comment]
Yes, I did include the `collectd` recipe before the `attribute_driven` one

**Comment by hectcastro** [comment]
Hm. The `run_list` for this cookbook's Test Kitchen [definition](https://github.com/hectcastro/chef-collectd/blob/3b66887747fd02fb7437059c692ab97007cc2ca7/.kitchen.yml#L21-L23) includes both recipes and all tests pass on Ubuntu and CentOS. The Chef client version is `12.4.1`.

Any more details around how you're pulling the cookbook in? What the `run_list` looks like?

**Comment by sterago** [comment]
I have declared the collectd cookbook dependency in the Berksfile like this

```
cookbook 'collectd-ng', '~> 1.2.0'
```

and, similarly, in the metadata.rb:

```
depends 'collectd-ng', '~> 1.2.0'
```

Then I defined a role in JSON format, referencing the two collectd recipes:

```
{
  "name": "collectd",
  "description": "Collectd basic configuration",
  "default_attributes": {
    "collectd": {
      "plugins": {
        "syslog": {
          "config": {"LogLevel": "Info"}
        },
        "disk": {},
        "swap": {},
        "memory": {},
        "cpu": {}
      }
    }
  },
  "override_attributes": {
  },
  "chef_type": "role",
  "run_list": [
    "recipe[collectd-ng]",
    "recipe[collectd-ng::attribute_driven]"
  ],
  "env_run_lists": {
  }
}
```

the .kitchen.yml file references the collectd role in this way:

```
    run_list:
      - role[collectd]

```

**Comment by hectcastro** [comment]
Ah, the name of the LWRP for the version of this cookbook that gets installed via Chef Supermarket is `collectd_ng_plugin`. Names have changed because of the Chef Supermarket namespacing. Some more details around the changes can be found in https://github.com/hectcastro/chef-collectd/issues/45.

**Comment by adampats** [comment]
I am running into this same error.  @hectcastro what does your reference to `collectd_ng_plugin` mean exactly?

For now I am skipping the use of the `attribute_driven` recipe and just providing my conf options in a template, but it would be nice to convert this to a role based set of attributes (I have multiple write_graphite prefixes depending on server role).

**Comment by hectcastro** [comment]
> I am running into this same error. @hectcastro what does your reference to collectd_ng_plugin mean exactly?

Since there is already a `collectd` cookbook in the Chef Supermarket, I was only able to upload this cookbook as `collectd_ng`. The branch with those changes is [`feature/hmc/chef-supermarket`](https://github.com/hectcastro/chef-collectd/tree/feature/hmc/chef-supermarket`). It has not been merged because it has the potential to break usage for everyone referencing this cookbook via Git in their `Berksfile`.
